java/code/src/com/redhat/rhn/domain/channel/ChannelFactory.java

```diff
@@ -1040,6 +1040,22 @@ public static DistChannelMap lookupDistChannelMapByOrgReleaseArch(Org org,
                 "DistChannelMap.findByChannel", params);
     }
 
+    /**
+     * All channels (including children) accessible for the user, ordered by the following rules
+     *
+     * 1) Base channels are listed first
+     * 2) Parent channels are ordered by label
+     * 3) Child channels are listed right after the corresponding parent, and ordered by label
+     *
```

Comment:
One missing condition on this information. 
Child channels are included only if the user has access to the corresponding parent channel.

Reply:
Done